### ISSUE TITLE
Not obvious if ref float is two blocks or one

### ISSUE THREAD

**Issue by olleharstedt** [krohrer/caml-inspect#5]
Hi,

Using inspect, we found it a little unclear what to think about referenced floats. According to Obj, `ref 1.0` has tag 0, but `1.0` has tag 253 (double_tag). With inspect:

```
# Inspect.Sexpr.dump @@ ref 1.0 ;;
(DUMP (BLK/0 :TAG 0 :VALUES 1.000000))
```

Is `VALUES` to be understood as a block here? Maybe it should me made more apparent? In contrast:

```
# Inspect.Sexpr.dump @@ [| 1.2 |] ;;
(DUMP (DBLA/0 1.2))
```

No `VALUES` block for optimized double array.

**Comment by krohrer** [comment]
Yes, I agree that this is a little bit confusing. For the S-expression dumper, I decided to inline all the value types (`floats` / known custom values like `int32`,`int64`,`nativeint`), instead of printing them for what they are (Blocks with double_tag and custom values respectively).

Ideally, this would be customizable by the context, akin to what is done in the `Dot` module.

I am not too happy with the output format of the Sexpr module anyway, maybe I'll rework it a little bit in the near future. Right now, there really is no proper way to dump the exact contents of a value (assuming that there is an exact string representation thereof).

If you have any other suggestions, please let me know.
